### ISSUE TITLE
Bazel CI: Bazel watcher and rules_jvm_external broken with Bazel@HEAD

### ISSUE THREAD

**Issue by meteorcloudy** [bazelbuild/bazel#16746]
Bazel Auto Sheriff report: https://buildkite.com/bazel/bazel-auto-sheriff-face-with-cowboy-hat/builds/1093

Bazel Watcher (from `//internal/e2e/output_runner:output_runner_test`): 
```
output_runner_test.go:160: Expected 3 commands to be executed, but found 0.
```

rules_jvm_external:
```
FAILURE: Expected regexp "Assuming maven local for artifact: com.example:kt:1.0.0" not found
```
https://github.com/bazelbuild/rules_jvm_external/search?q=Assuming+maven+local+for+artifact

Culprit: <del>https://github.com/bazelbuild/bazel/commit/6c77c5a4ad82816b3e448364fc9168d9d4e1c44f</del> https://github.com/bazelbuild/bazel/commit/6e945e8ab422ca0a195d09899cce61ba6c0cfa59

To reproduce:
```
git clone https://github.com/bazelbuild/bazel-watcher.git
export USE_BAZEL_VERSION=6e945e8ab422ca0a195d09899cce61ba6c0cfa59
bazelisk test //internal/e2e/output_runner:output_runner_test
```

This cannot be reproduced at https://github.com/bazelbuild/bazel/commit/6e945e8ab422ca0a195d09899cce61ba6c0cfa59 parent commit

**Comment by meteorcloudy** [comment]
@justinhorvitz It looks like some projects may depend on Bazel's output, do you see any easy way to avoid breaking them? Otherwise, the downstream projects may have to adapt this change.

**Comment by meteorcloudy** [comment]
Ah, you are right, it probably should be https://github.com/bazelbuild/bazel/commit/6e945e8ab422ca0a195d09899cce61ba6c0cfa59, the culprit finder detected https://github.com/bazelbuild/bazel/commit/6c77c5a4ad82816b3e448364fc9168d9d4e1c44f because it failed to download the Bazel binary built at this commit:
```
:googlecloud: Downloading Bazel built at 6c77c5a4ad82816b3e448364fc9168d9d4e1c44f
 
gsutil cp gs://bazel-builds/artifacts/centos7/6c77c5a4ad82816b3e448364fc9168d9d4e1c44f/bazel /tmp/tmp5j9s4cu5/bazel
CommandException: No URLs matched: gs://bazel-builds/artifacts/centos7/6c77c5a4ad82816b3e448364fc9168d9d4e1c44f/bazel
Failed to download Bazel binary at 6c77c5a4ad82816b3e448364fc9168d9d4e1c44f, error message:
Command '['gsutil', 'cp', 'gs://bazel-builds/artifacts/centos7/6c77c5a4ad82816b3e448364fc9168d9d4e1c44f/bazel', '/tmp/tmp5j9s4cu5/bazel']' returned non-zero exit status 1.
 
```

I'll look into the infra issue

**Comment by meteorcloudy** [comment]
I have fixed the missing Bazel binary issue, and indeed it succeeded at https://github.com/bazelbuild/bazel/commit/6c77c5a4ad82816b3e448364fc9168d9d4e1c44f and failed at https://github.com/bazelbuild/bazel/commit/6e945e8ab422ca0a195d09899cce61ba6c0cfa59.

**Comment by justinhorvitz** [comment]
Thanks for confirming the culprit and fixing the infra issue. Regarding the failing tests, can we determine whether the issue is an assumption in the test about `print()` semantics vs a real production issue? The difference is that prints are not replayed on incremental builds when the Starlark code is not executed. Can tests which run multiple `bazel` commands just check the output after the first command, or run `bazel clean` between the commands? See the test modifications I made in https://github.com/bazelbuild/bazel/commit/6e945e8ab422ca0a195d09899cce61ba6c0cfa59.

For more background on the change: on https://github.com/bazelbuild/bazel/issues/16721, the issue reporter is asking for a cherrypick to cover up the regression in `print()` behavior for duplicate print statements. I decided not just to fix the regression, but to make a more principled fix to make `print()` operate more intuitively (also, the principled fix was actually much easier than trying to restore previous behavior). I was aware of the possibility of breaking stuff that depends on bazel output, but per https://bazel.build/rules/bzl-style#print-for-debugging, "Do not use the `print()` function in production code" gave me more confidence to proceed with the change.

My hope is that we can keep the change and just clean up the tests. @meteorcloudy whose responsibility is it to investigate that?
